src/backends/gaston.jl

```diff
@@ -0,0 +1,293 @@
+# https://github.com/mbaz/Gaston.
+const G = Gaston
+const GastonSubplot = G.Plot
+const GASTON_MARKER_SCALING = 1.3 / 5.0
+const GNUPLOT_DPI = 72  # Compensate for DPI with increased resolution
+
+#
+# --------------------------------------------
+# These functions are called by Plots
+# --------------------------------------------
+#
+# Create the window/figure for this backend.
+function _create_backend_figure(plt::Plot{GastonBackend})
+    xsize = plt.attr[:size][1]
+    ysize = plt.attr[:size][2]
+    G.set(termopts="""size $xsize,$ysize""")
+
+    state_handle = G.nexthandle() # for now all the figures will be kept
+    plt.o = G.newfigure(state_handle)
+end
+
+
+function _before_layout_calcs(plt::Plot{GastonBackend})
+    # Initialize all the subplots first
+    plt.o.subplots = G.SubPlot[]
+    grid = size(plt.layout)
+    plt.o.layout = grid
+
+
+    for sp in plt.subplots
+        gaston_init_subplot(plt, sp)
+    end
+
+    # Then add the series (curves in gaston)
+    for series in plt.series_list
+        gaston_add_series(plt, series)
+    end
+end
+
+function _update_min_padding!(sp::Subplot{GastonBackend})
+    sp.minpad = (20mm, 5mm, 2mm, 10mm)
+end
+
+function _update_plot_object(plt::Plot{GastonBackend})
+end
+
+for (mime, term) in (
+    "application/eps"         => "epscairo",   # NEED fixing TODO
+    "image/eps"               => "epslatex",   # NEED fixing TODO
+    "application/pdf"         => "pdfcairo",   # NEED fixing TODO
+    "application/postscript"  => "postscript", # NEED fixing TODO
+    "image/svg+xml"           => "svg",
+    "text/latex"              => "tikz",       # NEED fixing TODO
+    "application/x-tex"       => "epslatex",   # NEED fixing TODO
+    "text/plain"              => "dumb",       # NEED fixing TODO
+)
+    @eval function _show(io::IO, ::MIME{Symbol($mime)}, plt::Plot{GastonBackend})
+        xsize = plt.attr[:size][1]
+        ysize = plt.attr[:size][2]
+        termopts="""size $xsize,$ysize"""
+
+        tmpfile = G.tempname() * "." * $term
+
+        G.save(
+            term = $term,
+            output = tmpfile,
+            handle = plt.o.handle,
+            saveopts = termopts
+        )
+        while !isfile(tmpfile) end  # avoid race condition with read in next line
+        write(io, read(tmpfile))
+        rm(tmpfile, force=true)
+        nothing
+
+    end
+end
+
+function _show(io::IO, mime::MIME{Symbol("image/png")},
```

Comment:
@mbaz Image/png I believe is for png figures, as gnuplot has a constant dpi I had add some scaling hacks here